### PR TITLE
New error kind and return code for indicating a trace is too long.

### PR DESCRIPTION
**Pull request by vext01**
I think this it roughly what we'd want to replace the assertion that we discussed in the previous PR.

While we are here, I was overloading `pte_overflow` in one place. We can use our new error kind to flag overflow errors and do away with a crusty utility function.

Looks good?

### CODE REVIEW

File: src/backends/perf_pt/collect.c [hunk 1]
```diff
@@ -178,8 +178,11 @@ read_aux(void *aux_buf, struct perf_event_mmap_page *hdr,
     if (required_capacity > trace->capacity) {
         // Over-allocate to 2x what we need, checking that the result fits in
         // the size_t argument of realloc(3).
-        // XXX replace assertion with a "trace too long" error kind.
-        assert(required_capacity < SIZE_MAX / 2);
+        if (required_capacity >= SIZE_MAX / 2) {
+            // The new size would overflow the size_t argument of realloc(3).
+            perf_pt_set_err(err, perf_pt_cerror_other, PERF_PT_TRACE_TOO_LONG);
```
Comment (ltratt) [inline src/backends/perf_pt/collect.c]: Ok, so I now think this isn't really about the "trace being too long", more "you asked us to allocate terrabytes of memory"?

Comment (ltratt) [reply src/backends/perf_pt/collect.c]: I didn't fully understand what you were getting at before. "Trace too long" is a misleading description IMHO: the trace isn't too long; rather, you're *asking* for the space for the trace to be ridiculously big. Is anyone ever going to hit this on a 64 bit machine? Nope. On a 32 bit machine... *maybe*... but probably not. And if someone was stupid enough to try allocating terrabytes of data, are they they going to be clever enough to reduce their request? Probably not ;) I think my original suggestion of `errx` might be the simplest way out of this.

Comment (vext01) [reply src/backends/perf_pt/collect.c]: For me it's either flag `ENOMEM` or as you prefer (?) `errx`. All I care is that the number isn't allowed to overflow. Modern linux no longer over-commits, so even `realloc` would return `ENOMEM` if there's not enough physical memory.

Comment (ltratt) [reply src/backends/perf_pt/collect.c]: Maybe we should just return `ENOMEM` in either case then?

**Comment by vext01** [comment]
Here we go. OK to squash?

**Comment by ltratt** [comment]
Works for me. Are you happy with it?

**Comment by vext01** [comment]
Fine by me.

**Comment by ltratt** [comment]
OK then, time to squash!

**Comment by vext01** [comment]
Splat.

**Pull request merged**